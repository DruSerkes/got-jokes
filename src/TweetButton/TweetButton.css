.TweetButton {
    background-image: url('./twitter_icon.png');
    background-position: center;
    background-repeat: no-repeat;
    background-size: contain;
    width: 70px !important;
}

