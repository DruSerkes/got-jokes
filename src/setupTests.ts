// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

export const testJoke = {
    "error": false,
    "category": 'dark',
    "type": 'single',
    "joke": 'test joke',
    "flags": {
        "nsfw": false,
        "religious": false,
        "political": false,
        "racist": false,
        "sexist": false,
        "explicit": false
    },
    "id": 13,
    "safe": true,
    "lang": 'en',
    'setup': 'test setup',
    'delivery': 'test delivery'
};

export const setJoke = jest.fn(() => console.log('setting joke'));
export const setViewingFavorites = jest.fn(() => console.log('viewing favorites'));
export const setFavorites = jest.fn(() => console.log('setting favorites'));
export const clearFavorites = jest.fn(() => console.log('clearing favorites'));
export const saveFavoriteJoke = jest.fn(() => console.log('saving favorite joke'));