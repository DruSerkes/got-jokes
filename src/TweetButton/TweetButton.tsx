import * as React from 'react';
import Button from 'react-bootstrap/Button';
import { jokeData } from '../types';
import './TweetButton.css';

interface TweetButtonProps {
  joke: jokeData
};

export const TweetButton = ({ joke }: TweetButtonProps) => {
  const tweetText = joke.joke ? joke.joke : `${joke.setup} ${joke.delivery}`;
  const tweetLink = `https://twitter.com/intent/tweet?text=${tweetText}`;
  return (
    <Button
      as="a"
      variant="outline-dark"
      title="tweet this joke"
      href={tweetLink}
      target="_blank"
      rel="noopener noreferrer"
      className='my-4 TweetButton'
    />
  );
};