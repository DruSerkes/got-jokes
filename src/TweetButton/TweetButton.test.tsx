import * as React from 'react';
import { render } from '@testing-library/react';
import { TweetButton } from './TweetButton';
import { testJoke } from '../setupTests';

describe('TweetButton tests', () => {

    it('should render and match snapshot', () => {
        const { asFragment } = render(<TweetButton joke={testJoke} />);
        expect(asFragment()).toMatchSnapshot();
    });

}); 