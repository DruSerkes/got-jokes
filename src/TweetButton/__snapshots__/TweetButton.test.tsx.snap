// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetButton tests should render and match snapshot 1`] = `
<DocumentFragment>
  <a
    class="my-4 TweetButton btn btn-outline-dark"
    href="https://twitter.com/intent/tweet?text=test joke"
    rel="noopener noreferrer"
    target="_blank"
    title="tweet this joke"
  />
</DocumentFragment>
`;
